packages/plugin-npm/sources/npmHttpUtils.ts

```diff
@@ -0,0 +1,34 @@
+import {Configuration, httpUtils, Ident} from '@berry/core';
+
+export function get(url: string, ident: Ident, configuration: Configuration) {
+  const headers: {[headerName: string]: string} = {};
+
+  const auth = getAuthenticationHeader(ident, configuration);
+  if (auth)
+    headers.authorization = auth;
+
+  return httpUtils.get(
+      url,
+      configuration,
+      {headers},
+  );
+}
+
+function getAuthenticationHeader(ident: Ident, configuration: Configuration) {
+  const mustAuthenticate = configuration.get(`npmAlwaysAuth`);
+
+  if (!mustAuthenticate && !ident.scope)
+    return null;
+
+  if (configuration.get(`npmAuthToken`))
+    return `Bearer ${configuration.get(`npmAuthToken`)}`;
+
+  if (configuration.get(`npmAuthIdent`))
+    return `Basic ${configuration.get(`npmAuthIdent`)}`;
+
+  if (mustAuthenticate) {
+    throw new Error(`No authentication configured for request`);
```

Comment:
Can you add a new message code to `Report.ts` and throw a `ReportError`?

Reply:
```
$ yarn add is-ci@latest
➤ YN0000: ┌ Resolution step
➤ YN0033: │ is-ci@npm:latest: No authentication configured for request
➤ YN0000: └ Completed in 0.2s
➤ YN0000: Failed with errors in 0.2s
```

Reply:
I see no changes to `Report.ts` or `ReportError`, am I missing a change? 🤔

Reply:
Oops, that's what you get when you start ignoring the output of `git push`. Pushed!